h2/stream.py

```diff
@@ -684,6 +691,9 @@ def __init__(self, stream_id):
         # The authority we believe this stream belongs to.
         self._authority = None
 
+        # The configuration for this stream.
+        self.config = H2Configuration()
```

Comment:
Don't create a brand new config object. In fact, ignore my prior comment: we should probably provide this on the constructor, as we cannot reasonably expect streams to exist without it.